framework/src/utils/MeshBaseDiagnosticsUtils.C

```diff
@@ -66,4 +66,85 @@ checkNonConformalMesh(const std::unique_ptr<MeshBase> & mesh,
   }
   pl->unset_close_to_point_tol();
 }
+
+bool
+checkEdgeOverlap(const std::unique_ptr<Elem> & edge1,
+                 const std::unique_ptr<Elem> & edge2,
```

Comment:
you need to check that these elements are actually edges. use a mooseAssert

Reply:
I included a check that should return the element type and I have it fail if it's anything but 0. If we wanted something a bit more straightforward I can just have it check that get_nodes() returns 2 nodes? Let me know what you think.